compiler/bindings/python/CMakeLists.txt

```diff
@@ -176,13 +176,14 @@ set(_SOURCE_COMPONENTS
 
   MLIRPythonSources.Core
 
-  # Core dialects (constrained to IREE input dialects).
+  # Core dialects.
   MLIRPythonSources.Dialects.amdgpu
   MLIRPythonSources.Dialects.arith
   MLIRPythonSources.Dialects.builtin
   MLIRPythonSources.Dialects.cf
   MLIRPythonSources.Dialects.complex
   MLIRPythonSources.Dialects.func
+  MLIRPythonSources.Dialects.gpu
```

Comment:
This may be failing on Windows.

https://github.com/openxla/iree/actions/runs/8739216455/job/23980064606#step:8:8748
```
[7860/8173] Linking CXX shared module compiler\bindings\python\iree\compiler\_mlir_libs\_mlirGPUPasses.cp310-win_amd64.pyd
FAILED: compiler/bindings/python/iree/compiler/_mlir_libs/_mlirGPUPasses.cp310-win_amd64.pyd 
C:\Windows\system32\cmd.exe /C "cd . && "C:\Program Files\CMake\bin\cmake.exe" -E vs_link_dll --intdir=compiler\bindings\python\CMakeFiles\IREECompilerPythonModules.extension._mlirGPUPasses.dso.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\mt.exe --manifests  -- C:\PROGRA~1\MICROS~2\2022\ENTERP~1\VC\Tools\MSVC\1438~1.331\bin\HostX64\x64\link.exe /nologo compiler\bindings\python\CMakeFiles\IREECompilerPythonModules.extension._mlirGPUPasses.dso.dir\__\__\__\third_party\llvm-project\mlir\lib\Bindings\Python\GPUPasses.cpp.obj  /out:compiler\bindings\python\iree\compiler\_mlir_libs\_mlirGPUPasses.cp310-win_amd64.pyd /implib:compiler\bindings\python\iree\compiler\_mlir_libs\_mlirGPUPasses.lib /pdb:compiler\bindings\python\iree\compiler\_mlir_libs\_mlirGPUPasses.pdb /dll /version:0.0 /machine:x64 -fuse-ld=lld /debug /INCREMENTAL  -natvis:C:/a/iree/iree/runtime/iree.natvis  lib\IREECompiler.lib  C:\hostedtoolcache\windows\Python\3.10.11\x64\libs\python310.lib  C:\hostedtoolcache\windows\Python\3.10.11\x64\libs\python310.lib  llvm-project\lib\LLVMSupport.lib  psapi.lib  shell32.lib  ole32.lib  uuid.lib  advapi32.lib  ws2_32.lib  delayimp.lib  -delayload:shell32.dll  -delayload:ole32.dll  llvm-project\lib\LLVMDemangle.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib && cd ."
LINK Pass 1: command "C:\PROGRA~1\MICROS~2\2022\ENTERP~1\VC\Tools\MSVC\1438~1.331\bin\HostX64\x64\link.exe /nologo compiler\bindings\python\CMakeFiles\IREECompilerPythonModules.extension._mlirGPUPasses.dso.dir\__\__\__\third_party\llvm-project\mlir\lib\Bindings\Python\GPUPasses.cpp.obj /out:compiler\bindings\python\iree\compiler\_mlir_libs\_mlirGPUPasses.cp310-win_amd64.pyd /implib:compiler\bindings\python\iree\compiler\_mlir_libs\_mlirGPUPasses.lib /pdb:compiler\bindings\python\iree\compiler\_mlir_libs\_mlirGPUPasses.pdb /dll /version:0.0 /machine:x64 -fuse-ld=lld /debug /INCREMENTAL -natvis:C:/a/iree/iree/runtime/iree.natvis lib\IREECompiler.lib C:\hostedtoolcache\windows\Python\3.10.11\x64\libs\python310.lib C:\hostedtoolcache\windows\Python\3.10.11\x64\libs\python310.lib llvm-project\lib\LLVMSupport.lib psapi.lib shell32.lib ole32.lib uuid.lib advapi32.lib ws2_32.lib delayimp.lib -delayload:shell32.dll -delayload:ole32.dll llvm-project\lib\LLVMDemangle.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:compiler\bindings\python\CMakeFiles\IREECompilerPythonModules.extension._mlirGPUPasses.dso.dir/intermediate.manifest compiler\bindings\python\CMakeFiles\IREECompilerPythonModules.extension._mlirGPUPasses.dso.dir/manifest.res" failed (exit code 1120) with the following output:
LINK : warning LNK4044: unrecognized option '/fuse-ld=lld'; ignored
   Creating library compiler\bindings\python\iree\compiler\_mlir_libs\_mlirGPUPasses.lib and object compiler\bindings\python\iree\compiler\_mlir_libs\_mlirGPUPasses.exp
LINK : warning LNK4199: /DELAYLOAD:shell32.dll ignored; no imports found from shell32.dll
LINK : warning LNK4199: /DELAYLOAD:ole32.dll ignored; no imports found from ole32.dll

GPUPasses.cpp.obj : error LNK2019: unresolved external symbol mlirRegisterGPUPasses referenced in function "void __cdecl pybind11_init__mlirGPUPasses(class pybind11::module_ &)" (?pybind11_init__mlirGPUPasses@@YAXAEAVmodule_@pybind11@@@Z)

compiler\bindings\python\iree\compiler\_mlir_libs\_mlirGPUPasses.cp310-win_amd64.pyd : fatal error LNK1120: 1 unresolved externals
```

Reply:
I can repro on my machine.

Reply:
Fix: https://github.com/openxla/iree/pull/17090